src/Caching/Cache.php

```diff
@@ -98,6 +98,45 @@ public function load($key, $fallback = NULL)
 
 
 	/**
+	 * Reads multiple items from the cache
+	 * @param array
+	 * @param callable
+	 * @return array
+	 */
+	public function bulkLoad(array $keys, $fallback = NULL)
+	{
+		foreach ($keys as $key) {
+			if (!is_scalar($key)) {
+				throw new Nette\InvalidArgumentException('Only scalar keys are allowed in a bulkLoad method.');
+			}
+		}
+		$keys = array_combine(array_map([$this, 'generateKey'], $keys), $keys);
+		$storage = $this->getStorage();
+		if ($storage instanceof IBulkReadStorage) {
+			$cacheData = $storage->bulkRead(array_keys($keys));
+		} else {
+			$cacheData = [];
+			foreach ($keys as $storageKey => $key) {
+				$cacheData[$storageKey] = $this->load($key);
+			}
+		}
+		$result = [];
+		foreach ($keys as $storageKey => $key) {
+			if (isset($cacheData[$storageKey])) {
```

Comment:
I think it can't. I need to rearrange to the right order, reindex items and fill missing items with null.